### PR TITLE
adds more info to SQUASHING.md including diagrams in docs/assets

### PR DESCRIPTION
**Pull request by rileytaylor**
# What's in this PR?

This adds more detailed information to SQUASHING.md, including:
- what a rebase is
- super fancy diagrams
- what squashing does
- a few more git config editor options (VS code being the superior of the additional options, of course)

I also added some more details based on feedback from onboarding new members and working through the squashing/rebase process.

## References
Progress on: #423

### CODE REVIEW

**Comment by JoshSmith** [comment]
@rileytaylor does this fix #561 ?

**Comment by rileytaylor** [comment]
@JoshSmith No, but #577 does. I just updated it to resolve the issue on merge.

File: docs/SQUASHING.md [hunk 1]
```diff
@@ -1,16 +1,68 @@
 # How do I "squash" my commits?
 
-Before you start, [make sure your text editor is associated with Git](https://help.github.com/articles/associating-text-editors-with-git/) to not close without a response, otherwise you'll see some weird behavior.
+## Why do I need to "squash"?
 
-Then update your refs to be sure you have the latest code:
+Squashing is important to reduce the number of commits that go into develop and keep the history simpler. Often a feature branch may include a number of large commits as well as a few smaller ones for minor edits (like spelling and formatting fixes). This is good: you should always commit often on your feature branch. However, once everything has been hashed out and your Pull Request has been approved, we do not necessarily need the history of ever commit and it's respective message. For this reason, we like to "squash" our commits into one before merging a Pull Request.
```
Comment (JoshSmith) [inline docs/SQUASHING.md]: Prefer first line as separate line. Ticks around `develop`. Lowercase p and r in "pull request".

File: docs/SQUASHING.md [hunk 2]
```diff
@@ -1,16 +1,68 @@
...
+
+The process of "squashing" your commits is done through a rebase.
+
+## What is a rebase?
+
+With large projects (such as ours), there are many different people working simultaneously. After submitting a pull request, chances are that someone else's pull request will be merged into develop before yours is, resulting in your feature branch being behind. When this happens, your branch goes "stale" and needs to be revitalized through a rebase. While stale, conflicts may occur if someone also edited a line you edited as well. Most of the time this is not the case, but either way, you will need to rebase before merging an approved pull request.
```
Comment (JoshSmith) [inline docs/SQUASHING.md]: - `develop`
- "yours" vs "yours is"
- remove "as well"
- remove last sentence

File: docs/SQUASHING.md [hunk 3]
```diff
@@ -21,32 +73,33 @@ git rebase -i develop
...
+With `n` being the number of commits you have added. Unsure of how many commits you've added? Check out the [`log` command.](https://www.git-scm.com/docs/git-log)
 
 If you are working from the main repo run:
 
 ```shell
 git rebase -i origin/develop
 ```
 
-This says to rebase your work onto develop.
+This says to rebase your work onto develop *from* the main repository.
```
Comment (JoshSmith) [inline docs/SQUASHING.md]: Prefer `develop`.

**Pull request merged**